RocketChat/Rocket.Chat
# feat: Allow granting the `mobile-upload-file` permission to guests

## Proposed changes (including videos or screenshots)

 - Allow granting the `mobile-upload-file` permission to the guest role (but not make it the default behavior)

## Issue(s)


## Steps to test or reproduce

Use the `authorization:addPermissionToRole` Meteor method to check that the `mobile-upload-file` permission can now be granted to the guest role. **Example (cURL) request:**

```
curl --request POST \
  --url http://localhost:3000/api/v1/method.call/authorization%3AaddPermissionToRole \
  --header 'Content-Type: application/json' \
  --header 'X-Auth-Token: jVrwkAmobgfne3aEdL-dJh-znNXMkhg79pEAZS2rEIg' \
  --header 'X-User-Id: rMq9ovRmZabXHW4Do' \
  --data '{"message":"{\"msg\":\"method\",\"id\":\"16\",\"method\":\"authorization:addPermissionToRole\",\"params\":[\"mobile-upload-file\",\"guest\"]}"}'
```

This Meteor method should return an error (`Permission is restricted [error-action-not-allowed]`) in case an invalid permission is provided (e.g. a permission that can't be granted to the guest role -- any permission that isn't in this list: `view-d-room`, `view-joined-room`, `view-p-room`, `start-discussion`, `mobile-upload-file`).

Use the `authorization:removeRoleFromPermission` Meteor method to check that the `mobile-upload-file` permission can also be removed from the guest role. **Example (cURL) request:**

```
curl --request POST \
  --url http://localhost:3000/api/v1/method.call/authorization%3AremoveRoleFromPermission \
  --header 'Content-Type: application/json' \
  --header 'X-Auth-Token: jVrwkAmobgfne3aEdL-dJh-znNXMkhg79pEAZS2rEIg' \
  --header 'X-User-Id: rMq9ovRmZabXHW4Do' \
  --data '{"message":"{\"msg\":\"method\",\"id\":\"16\",\"method\":\"authorization:addPermissionToRole\",\"params\":[\"mobile-upload-file\",\"guest\"]}"}'
```

## Further comments

[CORE-837]

[CORE-837]: https://rocketchat.atlassian.net/browse/CORE-837?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
